### PR TITLE
fix intermittent bug with 048_media_types_jsx

### PR DESCRIPTION
**Pull request by hayd**
This seemed to happen relatively frequently in CI, rather than sleep for N seconds we wait for the server to print something to stdout. This isn't a perfect solution but ought to suffice.

### CODE REVIEW

File: cli/test_util.rs [hunk 1]
```diff
@@ -62,16 +64,29 @@ pub fn http_server<'a>() -> HttpServerGuard<'a> {
...
+      let r = out.read_exact(&mut buf);
+      if r.is_ok() {
+        println!("tools/http_server.py ready");
+        break
+      }
+    }
+    std::thread::sleep(std::time::Duration::from_millis(100));
+    if i == 99 {
+      panic!("tools/http_server.py failed to start");
+    }
```
Comment (ry) [inline cli/test_util.rs]: Try this code, it is race free:

```rust
  let stderr = child.stderr.as_mut().unwrap();
  let mut lines = BufReader::new(stderr).lines();
  let line = lines.next().unwrap().unwrap();
  assert!(line.starts_with("ready"));
  // Now the http server is usable.
```

File: tools/http_server.py [hunk 2]
```diff
@@ -178,20 +178,20 @@ def spawn():
...
-               double_redirects_server(), inf_redirects_server())
-    try:
-        while all(s.thread.is_alive() for s in servers):
-            sleep(10)
-    except KeyboardInterrupt:
-        pass
+    with spawn() as servers:
+        try:
+            while all(s.thread.is_alive() for s in servers):
+                sleep(10)
```
Comment (ry) [inline tools/http_server.py]: Can we reduce this to 1?

File: cli/test_util.rs (hunk 1 above)
Comment (hayd) [reply cli/test_util.rs]: what if the server is never ready? will it throw?

File: tools/http_server.py (hunk 2 above)
Comment (hayd) [reply tools/http_server.py]: yes. This was mainly for use of server independently that it'll show an error if any of the servers crash. If it's manually killed it, e.g. by tests, it shouldn't be an issue.

File: cli/test_util.rs (hunk 1 above)
Comment (ry) [reply cli/test_util.rs]: No, it will hang - but I believe there are timeouts in the rust test runner.

Comment (hayd) [reply cli/test_util.rs]: I think it just says "048_media_types_jsx has been running over 60 seconds" it doesn't cancel. Though there is the one hour time out 😬

Comment (ry) [reply cli/test_util.rs]: Might need to flush the stdout in python?

Comment (hayd) [reply cli/test_util.rs]: That's the -u above, in the python Command. Not sure if necessary but I added it.

Looks like your update is good! (It errors if the socket is in use.)

File: tools/http_server.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def spawn():
     while any(not s.thread.is_alive() for s in servers):
         sleep(0.01)
     try:
-        print("ready")
+        print "ready"
```
Comment (hayd) [inline tools/http_server.py]: 😳

Comment (ry) [reply tools/http_server.py]: so it doesn't print parens when executed in 2.7

Comment (hayd) [reply tools/http_server.py]: Not true:
```
% python2 -c 'print("hello")' 
hello
```
Anyways, no big deal. We are ways away from python3! Once v8 builds with it then we can care.

**Pull request merged**